### PR TITLE
Fix editing blocks in studio

### PR DESCRIPTION
**Pull request by balderdash**
This has been broken for a while. I tried resetting to old commits to find when it first broke, but I can't get our javascript to build when I go back more than 6 weeks. I'm not sure how it worked in the first place.

So I just added an `onPuzzleComplete()` call when you hit run and `level.edit_blocks` is set.

### CODE REVIEW

File: apps/src/sites/studio/pages/initApp.js [hunk 1]
```diff
@@ -70,7 +70,7 @@ window.apps = {
         $(document).trigger('appInitialized');
       },
       onAttempt: function (report) {
-        if (appOptions.level.isProjectLevel) {
+        if (appOptions.level.isProjectLevel && !appOptions.level.edit_blocks) {
```
Comment (Hamms) [inline apps/src/sites/studio/pages/initApp.js]: why is this necessary in addition to the change to `runButtonClick`?

Comment (balderdash) [reply apps/src/sites/studio/pages/initApp.js]: For project levels, the attempt isn't reported until you hit the finish button. We don't show the finish button when editing project level templates (e.g. "New Play Lab Project"), so it needs to report the attempt here to be able to save changes.
